benchmark/worker/benchmark_client.go

```diff
@@ -83,12 +83,10 @@ func printClientConfig(config *testpb.ClientConfig) {
 }
 
 func setupClientEnv(config *testpb.ClientConfig) {
-	// Use one cpu core by default.
+	// Use all cpu cores available on machine by default.
 	// TODO: Revisit this for the optimal default setup.
-	if config.CoreLimit > 1 {
+	if config.CoreLimit > 0 {
 		runtime.GOMAXPROCS(int(config.CoreLimit))
```

Comment:
So one issue with using the limit rather than the list is that the client and server could end up using the same cores if it's running on the same machine. But that should be less of an issue now that all of our performance experiment tests are on different machines. I would suggest considering adding a TODO about using core list if there's any scope for doing so in go.